drivers/cc110x/cc110x_chanmap_433mhz_50khz_alt.c

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2018 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup   drivers_cc110x
+ * @{
+ *
+ * @file
+ * @brief     Alternative channel map implementing 50 kHz wide channels by
+ *            combining 2 LPD433 channels (each 25 kHz wide) and a channel
+ *            distance of 200 kHz.
+ *
+ * This structure is put in a separate file so that the linker can elide it when
```

Comment:
The linker should elide it in any case if it is not referenced.

Reply:
Lets say I have three files:

foo.c:
``` C
int foo = 42;
```
foobar.c:
``` C
int foo = 42, bar = 42;
```

and main.c:
``` C
#include "stdio.h"

extern int foo;

int main(void) {
	printf("Foo is %i\n", foo);
}
```

Now I'm compiling and linking with:
``` shell
gcc -c foo.c
gcc -c foobar.c
gcc -c main.c

gcc -o foo foo.o main.o
gcc -o foobar foobar.o main.o
```

Than this gives me:
``` shell
 ls -l foo foobar
-rwxr-xr-x    1 maribu   maribu       18688 Nov  6 17:17 foo
-rwxr-xr-x    1 maribu   maribu       18736 Nov  6 17:17 foobar
```

So foobar is bigger, even though `bar` is never referenced.

Reply:
Most of it is debug info. But still 8 bytes more when having unused data in a c file:

``` shell
$ strip foo
$ strip bar
$ ls -l foo foobar
-rwxr-xr-x    1 maribu   maribu       13992 Nov  6 17:24 foo
-rwxr-xr-x    1 maribu   maribu       14000 Nov  6 17:24 foobar
```

Reply:
RIOT compiles with function/data-sections and link time garbage collection, as do all size-concerned C users:

```
[kaspar@ng sizetest]$ export CFLAGS="-ffunction-sections -fdata-sections -Wl,-gc-sections"
[kaspar@ng sizetest]$ gcc $CFLAGS -c foo.c
[kaspar@ng sizetest]$ gcc $CFLAGS -c foobar.c
[kaspar@ng sizetest]$ gcc $CFLAGS -o foo foo.o main.o
[kaspar@ng sizetest]$ gcc $CFLAGS -o foobar foobar.o main.o
[kaspar@ng sizetest]$ ls -l foo foobar
-rwxr-xr-x 1 kaspar kaspar 16496 Nov  6 17:33 foo
-rwxr-xr-x 1 kaspar kaspar 16496 Nov  6 17:33 foobar
[kaspar@ng sizetest]$ 
```

I think having the structure in a different file is alright, but the notice is misleading.

Reply:
Hmm, maybe having all channel maps in one file and all configs in another file would be more readable, instead of one file per config and per chanmap. Especially in case more of them appear in the future.

Reply:
>  12 files changed, 169 insertions(+), 364 deletions(-)

:-) Thanks for pointing that out!